### DISCUSSION TITLE
links not working after update

### DISCUSSION THREAD

**Discussion by BeckeBauer** [kimai/kimai#4124] [On-Premise: Installation & Updates] [answered]
After updating from 1.x to 2.0.27 (including update from php8.0 to php8.2), most links aren't working anymore. For instance, the click on "System" in the left menu does not reveal the sub menu. When I enter `https://my.kimai/en/doctor` manually, the page is shown and also the sub menu belonging to "System". When I click on PHP (or any other row shown on the Doctor page), again nothing happens (since the sub menu is now shown, I can click, however, on "Roles" and the page opens - but clicking to modify the rights is ignored).

I checked the log but there is no entry. The only error message I could find is from the Console:
`Failed to find a valid digest in the 'integrity' attribute for resource 'https://my.kimai/build/app.1adbc54f.js' with computed SHA-384 integrity '6T9q9JZqHhsBoJZGvIHhZ+IW1Q1hNjpLQ8CZXjwIkNuHiEZBiSFbTkllKthgL9SR'. The resource has been blocked.`
and
`Uncaught ReferenceError: KimaiWebLoader is not defined at permissions:4107:28`

I tried with Nginx and Apache 2.4 but the result is the same

Any help is very much appreciated

**Comment by kevinpapst** [comment]
Kimai is not working without the Javascript.
The integrity hash must be sha384-5uMKKT4bInjtdsb3E6rH70YcbOenw11LI3aLD/dNdMAkoIIa4Pqu20RC3Sq0oLeO
Either some browser plugin is interfering or antivirus is changing the content or a file wasn't properly updated.
Check `git status` for the last point.

**Reply by kevinpapst** [reply, marked as answer]
See https://github.com/kimai/kimai/blob/2.0.26/public/build/entrypoints.json#L66

**Comment by BeckeBauer** [comment]
git status lists a lot of entries and ends:
![image](https://github.com/kimai/kimai/assets/51003649/a7f4c361-7ed6-4c49-bffe-11f9f8dfc94c)

I have not changed the browser I used before with Kimai 1.x and did not install / activated any antivirus programs; deactivating the built-in virus protection of Windows 11 did not help either - any other ideas where to look at?

**Comment by BeckeBauer** [comment]
I downgraded to Kimai 2.0.23 and everything works well. I also checked out the dockerized version Kimai 2.0.26 and it has the same issue (links not working). Were there any changes between 2.0.23 and 2.0.26 which could have caused the issue? 
(my test environment - where 2.0.23 is running and 2.0.26 is not - is exactly the same, same browser, same virus protection etc. so that it might have to do with changes of the Kimai code?)

**Reply by kevinpapst** [reply]
Does it work in the demo? 
If so: it is very likely your environment that causes the issue.

**Comment by BeckeBauer** [comment]
It is very likely that it is my environment. Therefore, I am trying to find out what I need to change in my environment. I just thought that if there where a change of code, which could potentially conflict with environment settings (despite the demo environment having no issue), it would help my to identify where I need to look. 

If someone else runs into this issue, let me know if you found a solution. So far, I will stick with 2.0.23

**Reply by kevinpapst** [reply]
It is easier to follow if you (w/c)ould reply to the matching answer nd not always open a new thread.

I have sent a link above to the file that contains the sha integrity. If this is the same in your code, but still the browser sees another integrity key, than you have a weird caching issue. Opcache? Did the file not update?

**Reply by BeckeBauer** [reply]
so much to learn... sorry for not using the reply field (wasn't aware that using the answer field provided below opens a new thread)

I understand now that there were changes of files within the sub-directory of public; I have manually deleted its content and cloned it again which did the tick - now, also the entrypoints.json file is up to date and the issue is gone.

I have no idea why it wasn't shown when executing git status - so I marked your hint in this regard as answer - thank you!
